File: javascript/Array_Hash/217_Contains_Duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const hashset = new Set();

    for (n of nums) {
        if (hashset.has(n)) return true;

        hashset.add(n);
    }
    return false;
};

File: python/Array_Hash/217_Contains_Duplicate.py
'''
Brute force:
Time: O(n^2)
Space: O(1)
Sorting:
Time: O(nlogn)  logn: sort
Space: O(1)
HashSet (insert: O(1)):
Time: O(n)
Space: O(n)
'''

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        hashset = set()

        for n in nums:
            if n in hashset:
                return True

            hashset.add(n)

        return False
